PcapDotNet/src/PcapDotNet.Packets/Dhcp/DhcpLayer.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using PcapDotNet.Base;
+using PcapDotNet.Packets.Arp;
+using PcapDotNet.Packets.Dhcp.Options;
+using PcapDotNet.Packets.Ethernet;
+using PcapDotNet.Packets.IpV4;
+
+namespace PcapDotNet.Packets.Dhcp
+{
+    /// <summary>
+    /// Represents a DHCP layer.
+    /// <seealso cref="DhcpDatagram"/>
+    /// </summary>
+    public sealed class DhcpLayer : SimpleLayer, IEquatable<DhcpLayer>
+    {
+        /// <summary>
+        /// Message op code
+        /// </summary>
+        public DhcpMessageType MessageType { get; set; }
+
+        /// <summary>
+        /// Hardware address type
+        /// </summary>
+        public ArpHardwareType HardwareType { get; set; }
+
+        /// <summary>
+        /// Hardware address length
+        /// </summary>
+        public byte HardwareAddressLength { get; set; }
```

Comment:
I would separate it from the ClientHardwareAddress to allow setting all ClientHardwareAddress-Bytes.
ex. HardwareAddressLength = 6 but the user still wants to set byte 7 - 16

Maybe automatically set the HardwareAddressLength to 6 if the ClientMacAddress is set. What do you think about that?